packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -64,6 +69,37 @@ export function TelemetryView(): React.ReactElement {
 		ITimestampedTelemetryEvent[] | undefined
 	>();
 	const [maxEventsToDisplay, setMaxEventsToDisplay] = React.useState<number>(DEFAULT_PAGE_SIZE);
+	const [selectedCategory, setSelectedCategory] = useState("");
+	const [filteredTelemetryEvents, setFilteredTelemetryEvents] = React.useState<
+		ITimestampedTelemetryEvent[] | undefined
+	>();
+	const [customSearch, setCustomSearch] = React.useState("");
```

Comment:
It would be good to document the semantics of each of these pieces of search / filter state. In particular how they are used (exact matches? partial matches? What does empty string / array mean? etc.).

Reply:
Also, if there is any state that we are tracking here that is strictly derived from other state and input props, we probably don't need to store it as state (we will have to recompute it any time the other state / props change anyways, so we don't really need to store it). Not sure if that's the case here or not, but figured I would call it out as a best practice,

Reply:
Re-opening, because I think the other pieces of state could use docs too. `eventNameOptions` and `matchingOptions`, for example, aren't completely evident based on their variable names. It would be nice to document how they are used, and what default states (i.e. `[]`) mean.